src/charm.py

```diff
@@ -154,6 +173,44 @@ def main(self, event):
 
         self.model.unit.status = ActiveStatus()
 
+    def _on_sidepanel_relation_changed(self, event: RelationChangedEvent):
+        try:
+            self._check_leader()
+        except CheckFailed as check_failed:
+            self.model.unit.status = check_failed.status
+            return
+        if event.app.name in SIDEBAR_EXTRA_OPTIONS:
+            self.log.info(f"ADDING {event.app.name} to side panel")
+            side_bar_tabs_dict = json.loads(self._stored.side_bar_tabs)
+            if (
+                SIDEBAR_EXTRA_OPTIONS[event.app.name]["menuLink"]
```

Comment:
The app name is configurable by the user, you can deploy katib with whatever name, how will this be reflected here?

Reply:
I think this relates to a broader conversation [here](https://github.com/canonical/kubeflow-dashboard-operator/pull/33#discussion_r931210113).  In general, dashboard shouldn't know anything ahead of time about the applications placed into the sidebar